pkg/controller/infinispan/autoscale.go

```diff
@@ -80,15 +74,22 @@ func autoscalerLoop(clusterNsn types.NamespacedName, r *ReconcileInfinispan) {
 		podList := &corev1.PodList{}
 		kubernetes.GetK8sResources(ispn.ObjectMeta.Name, ispn.ObjectMeta.Namespace, podList)
 
+		user := constants.DefaultOperatorUser
+		pass, err := kubernetes.GetPassword(user, ispn.Spec.Security.EndpointSecretName, ispn.Namespace)
+		if err != nil {
+			continue
+		}
+
+		cluster := ispnutil.NewCluster(user, pass, podList.Items[0].Namespace, ispn.GetEndpointScheme(), kubernetes)
 		for _, pItem := range podList.Items {
 			// time.Sleep(time.Duration(10000/len(podList.Items)) * time.Millisecond)
 			if metricMinPodNum == 0 {
-				metricMinPodNum, err = getMetricMinPodNum(consts.DefaultOperatorUser, pass, string(protocol), pItem)
+				metricMinPodNum, err = getMetricMinPodNum(pItem, cluster)
```

Comment:
Looks like we can arrange with common pod exec functions just providing pItem,Name instead of transferring whole `corev1.Pod` record